src/object/srv_obj_migrate.c

```diff
@@ -570,14 +571,190 @@ migrate_fetch_update_inline(struct migrate_one *mrone, daos_handle_t oh,
 	return rc;
 }
 
+static int
+obj_ec_encode_buf(daos_obj_id_t oid, struct daos_oclass_attr *oca,
+		  unsigned char *buffer, unsigned char *p_bufs[])
+{
+	struct obj_ec_codec	*codec;
+	unsigned int	cell_bytes = obj_ec_cell_rec_nr(oca);
+	unsigned int	k = obj_ec_data_tgt_nr(oca);
+	unsigned int	p = obj_ec_parity_tgt_nr(oca);
+	unsigned char	*data[k];
+	int		i;
+
+	codec = obj_ec_codec_get(daos_obj_id2class(oid));
+	D_ASSERT(codec != NULL);
+
+	for (i = 0; i < p && p_bufs[i] == NULL; i++) {
+		D_ALLOC(p_bufs[i], cell_bytes);
+		if (p_bufs[i] == NULL)
+			return -DER_NOMEM;
+	}
```

Comment:
just to confirm that no case require the possible data copy for fragmented buffer?

Reply:
no, this is using single contiguous buffer to fetch data.